tests/new_upgrades/test_provisioningtemplate.py

```diff
@@ -0,0 +1,161 @@
+"""Test for ProvisioningTemplates related Upgrade Scenario's
+
+:Requirement: UpgradedSatellite
+
+:CaseAutomation: Automated
+
+:CaseComponent: ProvisioningTemplates
+
+:Team: Rocket
+
+:CaseImportance: High
+
+"""
+
+from box import Box
+from fauxfactory import gen_alpha, gen_mac, gen_string
+import pytest
+
+from robottelo.config import settings
+from robottelo.constants import (
+    DEFAULT_ARCHITECTURE,
+    DEFAULT_PTABLE,
+    DEFAULT_PXE_TEMPLATE,
+)
+from robottelo.utils.shared_resource import SharedResource
+
+provisioning_template_kinds = ['provision', 'PXEGrub', 'PXEGrub2', 'PXELinux', 'iPXE']
+# provisioning_template_kinds = ['provision', 'PXEGrub', 'PXEGrub2', 'PXELinux']
+
+
+@pytest.fixture
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+def provisioning_templates_setup(
+    content_upgrade_shared_satellite,
+    pxe_loader,
+    upgrade_action,
+):
+    """Verify that created host can read provisioning templates.
+
+    :steps:
+        1. Create host on Satellite and trying rendering provisioning templates.
+        2. Upgrade the Satellite to the next or latest version.
+
+    :expectedresults:
+        1. Provisioning templates for host can be rendered before upgrading to new version.
+
+    :parametrized: yes
+    """
+    target_sat = content_upgrade_shared_satellite
+    with SharedResource(target_sat.hostname, upgrade_action, target_sat=target_sat) as sat_upgrade:
+        test_name = f'provisioning_template_upgrade_{gen_alpha(length=8)}'
+        org = target_sat.api.Organization(name=f'{test_name}_org').create()
+        location = target_sat.api.Location(name=f'{test_name}_location').create()
+        arch = (
+            target_sat.api.Architecture()
+            .search(query={'search': f'name="{DEFAULT_ARCHITECTURE}"'})[0]
+            .read()
+        )
+        domain = target_sat.api.Domain().search(
+            query={'search': f'name={target_sat.hostname.partition(".")[-1]}'}
+        )[0]
+        ptable = target_sat.api.PartitionTable().search(
+            query={'search': f'name="{DEFAULT_PTABLE}"'}
+        )[0]
+        pxe_template = (
+            target_sat.api.ProvisioningTemplate()
+            .search(query={'search': DEFAULT_PXE_TEMPLATE})[0]
+            .read()
+        )
+        os = (
+            target_sat.api.OperatingSystem()
+            .search(
+                query={'search': f'name="{settings.supportability.content_hosts.default_os_name}"'}
+            )[0]
+            .read()
+        )
+        os.architecture.append(arch)
+        os.ptable.append(ptable)
+        os.provisioning_template.append(pxe_template)
+        os.update(['architecture', 'ptable', 'provisioning_template'])
```

Comment:
Arch, ptable, pxe_template should be already present on the server, so why we're updating it explicitely?

Reply:
I don't recall why I added this step, but I'll try running the test with it removed.